build.sh

```diff
@@ -188,6 +187,29 @@ __buildnative=true
 export TZ=UTC 
 export MONO_THREADS_PER_CPU=2000
 
+# Use uname to determine what the CPU is.
+CPUName=$(uname -p)
+case $CPUName in
+    x86_64)
+        __BuildArch=x64
+        ;;
+
+    armv7l)
+        echo "Unsupported CPU $CPUName detected, build might not succeed!"
```

Comment:
It should be one of the options. This is just setting up a sensible default for the BuildArch. It can be overwritten by command line option for cross-compilation - the code for that is there already.